dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/encryption/AuditCrypto.java

```diff
@@ -2109,6 +2094,69 @@ static final byte[] decrypt(byte[] mesg, byte[] key, String cipher, boolean useJ
         return tmpMesg;
     }
 
+    /**
+     * @param key
+     * @param cipher
+     * @return
+     * @throws InvalidKeyException
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     */
+    private static SecretKey constructSecretKey(byte[] key, String cipher) throws InvalidKeyException, NoSuchAlgorithmException, InvalidKeySpecException, NoSuchProviderException {
+        SecretKey sKey = null;
+        if (cipher.indexOf("AES") != -1) { //This code for FIPS 140-3
+            // 16 bytes = 128 bit key
+            sKey = new SecretKeySpec(key, 0, 16, "AES");
+        } else {
+            DESedeKeySpec kSpec = new DESedeKeySpec(key);
+            SecretKeyFactory kFact = null;
+
+            kFact = (CryptoUtils.isFips140_3Enabled()) ? SecretKeyFactory.getInstance(encryptAlgorithm) : SecretKeyFactory.getInstance(encryptAlgorithm, provider);
```

Comment:
```suggestion
            kFact = SecretKeyFactory.getInstance(encryptAlgorithm);
```
Since AES is the 140-3 scenario, we do not need to do any special logic.  This puts the code back to the way it was.

Reply:
ok